lib/src/model/keychain.dart

```diff
@@ -257,25 +294,54 @@ KeyPair deriveArchethicKeypair(
   String curve = 'ed25519',
   String pathSuffix = '',
 }) {
-  // Hash the derivation path
-  final sha256 = Digest('SHA-256');
-  final hashedPath = sha256.process(
-    Uint8List.fromList(
-      replaceDerivationPathIndex(derivationPath, pathSuffix, index).codeUnits,
-    ),
+  final extendedSeed = deriveServiceSeed(
+    seed,
+    derivationPath,
+    index,
+    pathSuffix: pathSuffix,
   );
 
-  final hmac = crypto_lib.Hmac(crypto_lib.sha512, seed);
-  final digest = hmac.convert(hashedPath);
-  final extendedSeed = Uint8List.fromList(digest.bytes.sublist(0, 32));
-
   return crypto.generateDeterministicKeyPair(
```

Comment:
Here you miss a condition. After you get the extendedSeed, you should have 2 keypair generation algo:
- If the derivationPath contains the index, then you can use `generateDeterministicKeypair`
- If the derivationPath does not contains the index, then you need to creation a derivation of the seed with the index using `crypto.deriveKeypair(seed, index)`

Reply:
right. I Fixed it